src/validation.js

```diff
@@ -86,19 +86,24 @@ export function validationResultsForField(fieldUri, options) {
   return validationResults;
 }
 
-export function validationTypesForField(fieldUri, options) {
+export function validationsForFieldWithType(fieldUri, options) {
   const { store, formGraph } = options;
   const validationConstraints = validationsForField(fieldUri, options).map(
     (t) => t.object
   );
 
-  const validationTypes = validationConstraints
-    .map(
-      (constraintS) =>
-        store.match(constraintS, RDF("type"), undefined, formGraph)[0]
-    ) //There must be a least once
-    .map((triple) => triple.object);
-  return validationTypes;
+  const validationsWithType = {};
+  validationConstraints.forEach((constraintS) => {
+    const type = store.match(constraintS, RDF("type"), undefined, formGraph)[0]
+      .object;
+    validationsWithType[constraintS] = type;
```

Comment:
This will `.toString` the rdflib nodes instances which is not what we want I think? Can we still use them after the serialization to look up data (for example that maxLength value)?

In any case, I think it might be safer for now to simply make the `validationsForField` util public and the addon can then look for what it needs that way? That would already remove the `validatedBy` vs `validations` lookup logic from the addon.

I'm not 100% convinced about this util yet, but I'm no expert.

Reply:
If you think it's useful 👍. I just thought that it might be overkill for the single use case in the addon, but maybe it isn't.

It doesn't really matter that much anyways since we maintain both packages ourselves so we could easily change things if needed.